src/vsphere_cpi/spec/integration/cluster_cloud_properties_spec.rb

```diff
@@ -165,4 +165,93 @@
       end
     end
   end
+  context 'when disk_pool specifies a datastore accessible from cluster defined in vm_type' do
+    before (:all) do
+      @datastore_pattern = fetch_and_verify_datastore('BOSH_VSPHERE_CPI_DATASTORE_IN_DATASTORE_CLUSTER', @cluster_name)
+      #cluster which has a disjoint datastore (not shared with any other cluster)
+      @disjoint_cluster = fetch_and_verify_cluster('BOSH_VSPHERE_CPI_SECOND_CLUSTER')
+      @disjoint_datastore = fetch_and_verify_datastore('BOSH_VSPHERE_CPI_SECOND_CLUSTER_DATASTORE', @disjoint_cluster)
+    end
+    let(:vm_type) do
+      {
+        'ram' => 512,
+        'disk' => 2048,
+        'cpu' => 1,
+        'datacenters' => [
+          {
+            'name' => @datacenter_name,
+            'clusters' => [
+              {
+                @disjoint_cluster => {}
+              }
+            ]
+          }
+        ]
+      }
+    end
+    let(:cpi) do
+      options = cpi_options(
+        'datacenters' => [
+          {
+            'name' => @datacenter_name,
+            'datastore_pattern' => @datastore_pattern,
+            'persistent_datastore_pattern' => @datastore_pattern,
+            'clusters' => [
+              {
+                @cluster_name => {}
+              },
+            ]
+          }
+        ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    let(:disjoint_cluster_cpi) do
+      options = cpi_options(
+        'datacenters' => [
+          {
+            'name' => @datacenter_name,
+            'datastore_pattern' => @datastore_pattern,
+            'persistent_datastore_pattern' => @datastore_pattern,
+            'clusters' => [
+              {
+                @disjoint_cluster => {}
+              },
+            ]
+          }
+        ]
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    let(:disk_pool) { { 'datastores' => [@disjoint_datastore] } }
+
+    it 'should place disk into datastore that belongs to the cluster defined in cloud config' do
```

Comment:
we should probably asset that stemcell is also copied into particular datastore as well